domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/AppConfigBasedAmPersistenceSwitch.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.ccd.definition.store.domain;
+
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceReadSource.FROM_AM;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceReadSource.FROM_CCD;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceWriteDestination.TO_AM;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceWriteDestination.TO_BOTH;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceWriteDestination.TO_CCD;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.InvalidPropertyException;
+
+import com.google.common.collect.Maps;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Named
+@Singleton
+public class AppConfigBasedAmPersistenceSwitch implements AmPersistenceSwitch {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppConfigBasedAmPersistenceSwitch.class);
+
+    private final Map<String, AmPersistenceWriteDestination> caseTypesToWriteModes = Maps.newHashMap();
+
+    private final Map<String, AmPersistenceReadSource> caseTypesToReadModes = Maps.newHashMap();
+
+    public AppConfigBasedAmPersistenceSwitch(final ApplicationParams appParams) {
```

Comment:
Seems to me a bit over engineered. Why not do it simpler like https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/data/caseaccess/AMSwitch.java ?

Reply:
This one contains a duplicate configuration guard agains app params, and when such guard is added, a code in the link above would repeat the same logic five times. hence a guard is implemented with a template method.

Reply:
Deault is CCD, since the system should behave just as it is now in the absence of any express configuration about a case type. Basically, default is the current in effect.

Reply:
👍